tests/tasks/core/test_core.py

```diff
@@ -71,6 +71,15 @@ def my_fn():
         assert t.__wrapped__ == my_fn
         assert not hasattr(FunctionTask, "__wrapped__")
 
+    def test_function_task_raises_attribute_error(self):
+        def my_fn():
+            """An example function"""
+            pass
+
+        t = FunctionTask(fn=my_fn)
+        with pytest.raises(AttributeError):
+            t.unknown_attribute
```

Comment:
This would have passed before, since you're not checking the error message. Something like this should suffice:

```suggestion
        with pytest.raises(AttributeError) as exc:
            t.unknown_attribute
            
        assert "unknown_attribute" in str(exc.value)
        assert "@prefect.task" in str(exc.value)
```